File: src/main/java/quebec/virtualite/backend/services/domain/database/CityRepository.java
package quebec.virtualite.backend.services.domain.database;

import org.springframework.data.repository.CrudRepository;
import quebec.virtualite.backend.services.domain.entities.CityEntity;

import java.util.List;
import java.util.Optional;

public interface CityRepository extends CrudRepository<CityEntity, Long>
{
    List<CityEntity> findAllByOrderByNameAscProvinceAsc();

    Optional<CityEntity> findByName(String name);
}

File: src/main/java/quebec/virtualite/backend/services/domain/repositories/CityRepository.kt
package quebec.virtualite.backend.services.domain.repositories

import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Component
import quebec.virtualite.backend.services.domain.entities.CityEntity

@Component
interface CityRepository : CrudRepository<CityEntity, Long>
{
    fun deleteByName(name: String)
    fun findAllByOrderByNameAscProvinceAsc(): List<CityEntity>
    fun findByName(name: String): CityEntity?
}
